root/etc/s6-overlay/s6-rc.d/init-mod-code-server-rust-install/run

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/with-contenv bash
+
+set -e
+
+RUST_BINS_PATH="/rust-bins"
+PACKAGES_TO_INSTALL_LIST="/mod-repo-packages-to-install.list"
+
+ARCH=$(uname -m)
+SUPPORTED_ARCHS=("x86_64" "aarch64")
+
+if [[ ! " ${SUPPORTED_ARCHS[@]} " =~ " ${ARCH} " ]]; then
+  echo "+++ unsupported architecture: $ARCH +++"
+  exit 1
+fi
+
+RUST_BINARY_ARCHIVE_PATH="$RUST_BINS_PATH/rust-$ARCH-gnu.tar.gz"
+
+grep_os_release() {
+  grep -q "$1" /etc/os-release && true || false
+}
+
+if [[ -f "$RUST_BINARY_ARCHIVE_PATH" ]]; then
+  echo "+++ unpacking rust tar +++"
+  tar -xzf "$RUST_BINARY_ARCHIVE_PATH" -C "$RUST_BINS_PATH" --strip-components=1
+
+  echo "+++ installing rust +++"
+  "$RUST_BINS_PATH/install.sh"
+  rm -rf "$RUST_BINS_PATH"
+
+  echo "+++ adding build tools +++"
+  if grep_os_release "debian"; then
+    echo "build-essential" >>"$PACKAGES_TO_INSTALL_LIST"
```

Comment:
Please add a check to see if it's already installed like here: https://github.com/linuxserver/docker-mods/blob/code-server-python3/root/etc/s6-overlay/s6-rc.d/init-mod-code-server-python3-add-package/run#L3

Mod tarballs can be updated even when the container is restarted, not just recreated so there are cases where the mod tarball exists, but the repo packages are already installed.

Reply:
https://github.com/linuxserver/docker-mods/pull/808/commits/17a69de8dac0032eacf7a1c5d84d93f510f92019#diff-22bc35d6f7fcd5baecdae1ae65c1d1b48f027b17e5583a0a46c490a5fe155a84R5-R9

Reply:
That won't catch the scenario I mentioned above. Actually, that part should be removed because it will prevent a rust update when the init downloads a newer version mod on a restart.

Reply:
Ok, I think you know what you mean, but dpkg check won't work, I think the current behavior should catch this, case. I changed the message:
https://github.com/linuxserver/docker-mods/blob/a180cf76cf81f69950590012103534532fb25552/root/etc/s6-overlay/s6-rc.d/init-mod-code-server-rust-install/run#L27

Reply:
The golang mod, does it this way too:
https://github.com/linuxserver/docker-mods/blob/code-server-golang/root/etc/s6-overlay/s6-rc.d/init-mod-code-server-golang-add-package/run

Reply:
To further clarify, there are two things that need installed, rust package, and the repo package. They need separate checks because the former can be updated and may need a reinstall on a container restart, whereas the latter only needs to be installed after a recreation.

So it's prudent to have a check for the bins for the rust package, and if there are, install them (could mean it was never installed, or was installed but has an update, either way, we need to install the bins). And have a separate dpkg check for the repo package and add to package install list as needed.

Golang mod also needs to be updated with the repo package check.

Reply:
Sorry, it has been a long day. I totally forgot about the packages. So it should suffice to make the package check outside independently of the rust package check, right?

Reply:
e.g.:

`#!/usr/bin/with-contenv bash

set -e

RUST_BINS_PATH="/rust-bins"
PACKAGES_TO_INSTALL_LIST="/mod-repo-packages-to-install.list"

ARCH=$(uname -m)
SUPPORTED_ARCHS=("x86_64" "aarch64")

if [[ ! " ${SUPPORTED_ARCHS[@]} " =~ " ${ARCH} " ]]; then
  echo "**** unsupported architecture: $ARCH ****"
  exit 1
fi

RUST_BINARY_ARCHIVE_PATH="$RUST_BINS_PATH/rust-$ARCH-gnu.tar.gz"

if ! dpkg -l | grep build-essential >/dev/null; then
  echo "**** installing build-essential ****"
  echo "build-essential" >>"$PACKAGES_TO_INSTALL_LIST"
else
  echo "**** build-essential already installed ****"
fi

if [[ -f "$RUST_BINARY_ARCHIVE_PATH" ]]; then
  echo "**** unpacking rust tar ****"
  tar -xzf "$RUST_BINARY_ARCHIVE_PATH" -C "$RUST_BINS_PATH" --strip-components=1

  echo "**** installing rust ****"
  "$RUST_BINS_PATH/install.sh"
  rm -rf "$RUST_BINS_PATH"
else
  echo "**** rust bin does not exist ****"
fi
`

Reply:
Yup, this looks great